coral/src/app/features/topics/details/settings/TopicSettings.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  Alert,
+  BorderBox,
+  Box,
+  Button,
+  PageHeader,
+  Typography,
+  useToast,
+} from "@aivenio/aquarium";
+import { useState } from "react";
+import { useMutation } from "@tanstack/react-query";
+import { useNavigate } from "react-router-dom";
+import { deleteTopic } from "src/domain/topic";
+import { useTopicDetails } from "src/app/features/topics/details/TopicDetails";
+import { parseErrorMsg } from "src/services/mutation-utils";
+import { TopicDeleteConfirmationModal } from "src/app/features/topics/details/settings/components/TopicDeleteConfirmationModal";
+
+function TopicSettings() {
+  const { topicName, environmentId, topicOverview } = useTopicDetails();
+
+  const isTopicOwner = topicOverview.topicInfoList[0].topicOwner;
+  const showDeleteTopic = topicOverview.topicInfoList[0].showDeleteTopic;
+
+  const navigate = useNavigate();
+  const toast = useToast();
+
+  const [showConfirmation, setShowConfirmation] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<string | undefined>();
+
+  const { mutate, isLoading } = useMutation(
+    (params: { deleteAssociatedSchema: boolean; remark: string | undefined }) =>
```

Comment:
Maybe it is a little more consistent to export the param types from `domain` and use them to type the mutation? https://github.com/aiven/klaw/pull/1325/files#diff-490be5d38546d5750a92fc3785e12c4150091cc504a97b4a2e243ce16bceb2e4R220

Reply:
The mutation is still using inline types instead of DeleteTopicPayload. Is it a choice to not use domain types in app context?

Reply:
No, that was an oversight, sorry. Pushing a change!

Reply:
Thanks for noticing ❤️